src/value.h

```diff
@@ -501,6 +502,32 @@ class Value
     static Value Empty() { return EmbeddedVectorType(nullptr); }
   };
 
+  class ObjectType {
+  protected:
+    struct ObjectObject;
+    struct ObjectObjectDeleter {
+      void operator()(ObjectObject* obj);
+    };
+
+  private:
+    explicit ObjectType(const shared_ptr<ObjectObject> &copy);
+
+  public:
+    shared_ptr<ObjectObject> ptr;
+    ObjectType(class EvaluationSession* session);
+    ObjectType clone() const;
+    const Value& get(const std::string& key) const;
+    void set(const std::string& key, Value&& value);
+    Value operator==(const ObjectType &v) const;
+    Value operator< (const ObjectType &v) const;
+    Value operator> (const ObjectType &v) const;
+    Value operator!=(const ObjectType &v) const;
+    Value operator<=(const ObjectType &v) const;
+    Value operator>=(const ObjectType &v) const;
+    const Value& operator[](const str_utf8_wrapper &v) const;
+    const std::vector<std::string> keys() const;
```

Comment:
The suggested from LGTM seems not ideal. I think it's probably better to return a `const &` instead of removing the `const`.

Reply:
Done, thanks.